src/Localization/Abstractions/src/IStringLocalizer.cs

```diff
@@ -40,6 +41,7 @@ public interface IStringLocalizer
         /// </summary>
         /// <param name="culture">The <see cref="CultureInfo"/> to use.</param>
         /// <returns>A culture-specific <see cref="IStringLocalizer"/>.</returns>
+        [Obsolete("Set the CurrentCulture and CurrentUICulture instead.")]
```

Comment:
I think this is too strict. Let's go with: "This method is obsolete. Use `CurrentCulture` and `CurrentUICulture` instead". And please use this patter for the rest of the messages.